<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>Form</class>
 <widget class="QWidget" name="Form">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>764</width>
    <height>548</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Form</string>
  </property>
  <layout class="QHBoxLayout" name="horizontalLayout_21">
   <item>
    <layout class="QVBoxLayout" name="verticalLayout_10">
     <item>
      <layout class="QHBoxLayout" name="horizontalLayout_17">
       <item>
        <widget class="QGroupBox" name="sm_setting_box">
         <property name="title">
          <string>步进电机设置</string>
         </property>
         <layout class="QHBoxLayout" name="horizontalLayout_7">
          <item>
           <layout class="QVBoxLayout" name="verticalLayout">
            <item>
             <layout class="QHBoxLayout" name="horizontalLayout" stretch="1,2">
              <item>
               <widget class="QLabel" name="label">
                <property name="text">
                 <string>步进电机选择:</string>
                </property>
                <property name="alignment">
                 <set>Qt::AlignmentFlag::AlignCenter</set>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QComboBox" name="sm_select">
                <property name="currentIndex">
                 <number>0</number>
                </property>
                <item>
                 <property name="text">
                  <string>sm0</string>
                 </property>
                </item>
                <item>
                 <property name="text">
                  <string>sm1</string>
                 </property>
                </item>
                <item>
                 <property name="text">
                  <string>sm2</string>
                 </property>
                </item>
               </widget>
              </item>
             </layout>
            </item>
            <item>
             <layout class="QHBoxLayout" name="horizontalLayout_2" stretch="1,2">
              <item>
               <widget class="QLabel" name="label_2">
                <property name="text">
                 <string>速度：</string>
                </property>
                <property name="alignment">
                 <set>Qt::AlignmentFlag::AlignCenter</set>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QLineEdit" name="sm_speed">
                <property name="whatsThis">
                 <string/>
                </property>
                <property name="text">
                 <string>1</string>
                </property>
                <property name="placeholderText">
                 <string>输入速度范围：[1,255]</string>
                </property>
               </widget>
              </item>
             </layout>
            </item>
            <item>
             <layout class="QHBoxLayout" name="horizontalLayout_3" stretch="1,2">
              <item>
               <widget class="QLabel" name="label_3">
                <property name="text">
                 <string>加速度：</string>
                </property>
                <property name="alignment">
                 <set>Qt::AlignmentFlag::AlignCenter</set>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QLineEdit" name="sm_acc">
                <property name="toolTip">
                 <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;设置优先级：&lt;/p&gt;&lt;p&gt;加速度；平缓加速；速度和转动方向&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
                </property>
                <property name="text">
                 <string/>
                </property>
                <property name="placeholderText">
                 <string>范围：[-10,10] 不输入表示不设置</string>
                </property>
               </widget>
              </item>
             </layout>
            </item>
            <item>
             <layout class="QHBoxLayout" name="horizontalLayout_4" stretch="1,2">
              <item>
               <widget class="QLabel" name="label_4">
                <property name="text">
                 <string>转动方向：</string>
                </property>
                <property name="alignment">
                 <set>Qt::AlignmentFlag::AlignCenter</set>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QComboBox" name="sm_dir">
                <item>
                 <property name="text">
                  <string>0</string>
                 </property>
                </item>
                <item>
                 <property name="text">
                  <string>1</string>
                 </property>
                </item>
               </widget>
              </item>
             </layout>
            </item>
            <item>
             <layout class="QHBoxLayout" name="horizontalLayout_5" stretch="1,2">
              <item>
               <widget class="QLabel" name="label_5">
                <property name="text">
                 <string>平缓加速至：</string>
                </property>
                <property name="alignment">
                 <set>Qt::AlignmentFlag::AlignCenter</set>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QLineEdit" name="sm_smooth_speed">
                <property name="text">
                 <string/>
                </property>
                <property name="placeholderText">
                 <string>范围：[1,254] 不输入表示不设置</string>
                </property>
               </widget>
              </item>
             </layout>
            </item>
            <item>
             <layout class="QHBoxLayout" name="horizontalLayout_6" stretch="2,1,1,1">
              <item>
               <spacer name="horizontalSpacer">
                <property name="orientation">
                 <enum>Qt::Orientation::Horizontal</enum>
                </property>
                <property name="sizeHint" stdset="0">
                 <size>
                  <width>40</width>
                  <height>20</height>
                 </size>
                </property>
               </spacer>
              </item>
              <item>
               <widget class="QPushButton" name="mat_btn">
                <property name="text">
                 <string>绘制图表</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QPushButton" name="reset_step_btn">
                <property name="text">
                 <string>重置步数</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QPushButton" name="send_btn">
                <property name="text">
                 <string>发送指令</string>
                </property>
               </widget>
              </item>
             </layout>
            </item>
           </layout>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <layout class="QVBoxLayout" name="verticalLayout_7">
         <item>
          <widget class="QGroupBox" name="serial_setting_box">
           <property name="title">
            <string>串口设置</string>
           </property>
           <layout class="QHBoxLayout" name="horizontalLayout_16">
            <item>
             <layout class="QHBoxLayout" name="horizontalLayout_15">
              <item>
               <layout class="QVBoxLayout" name="verticalLayout_5">
                <item>
                 <widget class="QPushButton" name="open_serial_btn">
                  <property name="text">
                   <string>打开串口</string>
                  </property>
                 </widget>
                </item>
                <item>
                 <widget class="QPushButton" name="close_serial_btn">
                  <property name="text">
                   <string>关闭串口</string>
                  </property>
                 </widget>
                </item>
                <item>
                 <widget class="QPushButton" name="exit_btn">
                  <property name="text">
                   <string>退出程序</string>
                  </property>
                 </widget>
                </item>
               </layout>
              </item>
              <item>
               <layout class="QVBoxLayout" name="verticalLayout_6">
                <item>
                 <layout class="QHBoxLayout" name="horizontalLayout_12" stretch="1,2">
                  <item>
                   <widget class="QLabel" name="label_6">
                    <property name="text">
                     <string>波特率：</string>
                    </property>
                    <property name="alignment">
                     <set>Qt::AlignmentFlag::AlignCenter</set>
                    </property>
                   </widget>
                  </item>
                  <item>
                   <widget class="QComboBox" name="baudrate_select">
                    <property name="currentIndex">
                     <number>3</number>
                    </property>
                    <item>
                     <property name="text">
                      <string>1200</string>
                     </property>
                    </item>
                    <item>
                     <property name="text">
                      <string>2400</string>
                     </property>
                    </item>
                    <item>
                     <property name="text">
                      <string>4800</string>
                     </property>
                    </item>
                    <item>
                     <property name="text">
                      <string>9600</string>
                     </property>
                    </item>
                   </widget>
                  </item>
                 </layout>
                </item>
                <item>
                 <layout class="QHBoxLayout" name="horizontalLayout_13" stretch="1,2">
                  <item>
                   <widget class="QLabel" name="label_7">
                    <property name="text">
                     <string>串口名：</string>
                    </property>
                    <property name="alignment">
                     <set>Qt::AlignmentFlag::AlignCenter</set>
                    </property>
                   </widget>
                  </item>
                  <item>
                   <widget class="QLineEdit" name="port_name">
                    <property name="text">
                     <string>COM3</string>
                    </property>
                   </widget>
                  </item>
                 </layout>
                </item>
                <item>
                 <layout class="QHBoxLayout" name="horizontalLayout_14" stretch="1,2">
                  <item>
                   <widget class="QLabel" name="label_8">
                    <property name="text">
                     <string>超时时间：</string>
                    </property>
                    <property name="alignment">
                     <set>Qt::AlignmentFlag::AlignCenter</set>
                    </property>
                   </widget>
                  </item>
                  <item>
                   <widget class="QComboBox" name="timeout_select">
                    <property name="currentIndex">
                     <number>1</number>
                    </property>
                    <item>
                     <property name="text">
                      <string>0.5s</string>
                     </property>
                    </item>
                    <item>
                     <property name="text">
                      <string>1s</string>
                     </property>
                    </item>
                    <item>
                     <property name="text">
                      <string>1.5s</string>
                     </property>
                    </item>
                    <item>
                     <property name="text">
                      <string>2s</string>
                     </property>
                    </item>
                   </widget>
                  </item>
                 </layout>
                </item>
               </layout>
              </item>
             </layout>
            </item>
           </layout>
          </widget>
         </item>
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout_11">
           <item>
            <widget class="QGroupBox" name="sm0_box">
             <property name="title">
              <string>电机0信息</string>
             </property>
             <layout class="QHBoxLayout" name="horizontalLayout_8">
              <item>
               <layout class="QVBoxLayout" name="verticalLayout_2">
                <item>
                 <widget class="QLabel" name="sm0_speed">
                  <property name="text">
                   <string>速度：0</string>
                  </property>
                 </widget>
                </item>
                <item>
                 <widget class="QLabel" name="sm0_acc">
                  <property name="text">
                   <string>加速度：0</string>
                  </property>
                 </widget>
                </item>
                <item>
                 <widget class="QLabel" name="sm0_dir">
                  <property name="text">
                   <string>转动方向：0</string>
                  </property>
                 </widget>
                </item>
                <item>
                 <widget class="QLabel" name="sm0_rest_step">
                  <property name="text">
                   <string>剩余步数：0</string>
                  </property>
                 </widget>
                </item>
               </layout>
              </item>
             </layout>
            </widget>
           </item>
           <item>
            <widget class="QGroupBox" name="sm1_box">
             <property name="title">
              <string>电机1信息</string>
             </property>
             <layout class="QHBoxLayout" name="horizontalLayout_10">
              <item>
               <layout class="QVBoxLayout" name="verticalLayout_4">
                <item>
                 <widget class="QLabel" name="sm1_speed">
                  <property name="text">
                   <string>速度：0</string>
                  </property>
                 </widget>
                </item>
                <item>
                 <widget class="QLabel" name="sm1_acc">
                  <property name="text">
                   <string>加速度：0</string>
                  </property>
                 </widget>
                </item>
                <item>
                 <widget class="QLabel" name="sm1_dir">
                  <property name="text">
                   <string>转动方向：0</string>
                  </property>
                 </widget>
                </item>
                <item>
                 <widget class="QLabel" name="sm1_rest_step">
                  <property name="text">
                   <string>剩余步数：0</string>
                  </property>
                 </widget>
                </item>
               </layout>
              </item>
             </layout>
            </widget>
           </item>
           <item>
            <widget class="QGroupBox" name="sm2_box">
             <property name="title">
              <string>电机2信息</string>
             </property>
             <layout class="QHBoxLayout" name="horizontalLayout_9">
              <item>
               <layout class="QVBoxLayout" name="verticalLayout_3">
                <item>
                 <widget class="QLabel" name="sm2_speed">
                  <property name="text">
                   <string>速度：0</string>
                  </property>
                 </widget>
                </item>
                <item>
                 <widget class="QLabel" name="sm2_acc">
                  <property name="text">
                   <string>加速度：0</string>
                  </property>
                 </widget>
                </item>
                <item>
                 <widget class="QLabel" name="sm2_dir">
                  <property name="text">
                   <string>转动方向：0</string>
                  </property>
                 </widget>
                </item>
                <item>
                 <widget class="QLabel" name="sm2_rest_step">
                  <property name="text">
                   <string>剩余步数：0</string>
                  </property>
                 </widget>
                </item>
               </layout>
              </item>
             </layout>
            </widget>
           </item>
          </layout>
         </item>
        </layout>
       </item>
      </layout>
     </item>
     <item>
      <layout class="QHBoxLayout" name="horizontalLayout_20" stretch="5,1">
       <item>
        <widget class="QGroupBox" name="groupBox_6">
         <property name="title">
          <string>日志栏</string>
         </property>
         <layout class="QHBoxLayout" name="horizontalLayout_18">
          <item>
           <widget class="QPlainTextEdit" name="log_plain">
            <property name="toolTip">
             <string/>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <layout class="QVBoxLayout" name="verticalLayout_9">
         <item>
          <widget class="QGroupBox" name="groupBox">
           <property name="title">
            <string>状态栏</string>
           </property>
           <layout class="QHBoxLayout" name="horizontalLayout_19">
            <item>
             <layout class="QVBoxLayout" name="verticalLayout_8">
              <item>
               <widget class="QLabel" name="tem_label">
                <property name="text">
                 <string>温度：0℃</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QLabel" name="light_label">
                <property name="text">
                 <string>光照：0</string>
                </property>
               </widget>
              </item>
             </layout>
            </item>
           </layout>
          </widget>
         </item>
         <item>
          <spacer name="verticalSpacer">
           <property name="orientation">
            <enum>Qt::Orientation::Vertical</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>20</width>
             <height>40</height>
            </size>
           </property>
          </spacer>
         </item>
        </layout>
       </item>
      </layout>
     </item>
    </layout>
   </item>
  </layout>
 </widget>
 <resources/>
 <connections/>
</ui>
